source/firstprogram-expanding-network.rst

```diff
@@ -181,159 +185,324 @@ With these values defined, ``create24HourSchedules`` can be made. However, when
 
 .. code-block:: scala 
 
-  package sir
-
-  import java.util.Date
-  import com.bharatsim.engine.ContextBuilder._
-  import com.bharatsim.engine._
-  import com.bharatsim.engine.actions.StopSimulation
-  import com.bharatsim.engine.basicConversions.decoders.DefaultDecoders._
-  import com.bharatsim.engine.basicConversions.encoders.DefaultEncoders._
-  import com.bharatsim.engine.dsl.SyntaxHelpers._
-  import com.bharatsim.engine.execution.Simulation
-  import com.bharatsim.engine.graph.ingestion.{GraphData, Relation}
-  import com.bharatsim.engine.graph.patternMatcher.MatchCondition._
-  import com.bharatsim.engine.listeners.{CsvOutputGenerator, SimulationListenerRegistry}
-  import com.bharatsim.engine.models.Agent
-  import com.bharatsim.engine.utils.Probability.biasedCoinToss
-  import com.bharatsim.examples.epidemiology.sir.InfectionStatus._
-  import com.typesafe.scalalogging.LazyLogging
-
-  object Main extends LazyLogging {
-    private val initialInfectedFraction = 0.01
+    case class Person(id: Long, age: Int, infectionState: InfectionStatus, infectionDay: Int, currentLocation: String = "") extends Agent {
 
-    private val myTick: ScheduleUnit = new ScheduleUnit(1)
-    private val myDay: ScheduleUnit = new ScheduleUnit(myTick * 24)
+Then, we add a behaviour to update the current location of the person at every step.
 
-    def main(args: Array[String]): Unit = {
-
-      var beforeCount = 0
-      val simulation = Simulation()
+.. code-block:: scala 
 
-      simulation.ingestData(implicit context => {
-        ingestCSVData("citizen10k.csv", csvDataExtractor)
-        logger.debug("Ingestion done")
-      })
+    private def getCurrentLocation(context: Context): Option[String] = {
+      val schedule = context.fetchScheduleFor(this).get
+      val currentStep = context.getCurrentStep
+      val placeType: String = schedule.getForStep(currentStep)
 
-      simulation.defineSimulation(implicit context => {
-        create24HourSchedules()
+      Some(placeType)
+    }
 
-        registerAction(
-          StopSimulation,
-          (c: Context) => {
-            getInfectedCount(c) == 0
+    private def updateCurrentLocation(context: Context): Unit = {
+      val currentLocationOption = getCurrentLocation(context)
+      currentLocationOption match {
+        case Some(x) => {
+          if (this.currentLocation != x) {
+            updateParam("currentLocation", x)
           }
+        }
+
+        case _ => 
+      }
+    }
+
+    addBehaviour(updateCurrentLocation)
+
+We then also need to update the ``checkForInfection`` function to only check for infections in the same location. We start by counting the total number of people in the Person's current location, and then count the number of infected people in the same location. Then, we modify our infection rate calculation to be based on the number of infected people in the same location.

```

Comment:
Add one line here explaining what would go wrong if we didn't do this (everyone linked to a node would be considered, even if they were home).